File: C++/contadorCaracteres.cpp
#include <iostream>
#include <string>
#include <cctype>

using namespace std;

int main() {
    system("chcp 65001 > nul");

    string oracion;
    int numeros = 0, espacios = 0, letras = 0, vocales = 0, consonantes = 0;

    cout << "Ingrese una oración: ";
    getline(cin, oracion);

    for (char c : oracion) {
        if (isdigit(c)) numeros++;
        else if (isspace(c)) espacios++;
        else if (isalpha(c)) {
            letras++;

            c = tolower(c);

            if (c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u')vocales++;
            else consonantes++;
        }
    }

    cout << "\nNúmeros: " << numeros;
    cout << "\nEspacios: " << espacios;
    cout << "\nLetras: " << letras;
    cout << "\nVocales: " << vocales;
    cout << "\nConsonantes: " << consonantes;
    cout << "\nCaracteres totales: " << letras + numeros;
}

File: python/contador-caracteres.py
oracion = input("Ingrese una oracion: ")

numeros, espacios, letras, vocales, consonantes = 0, 0, 0, 0, 0
copia = []

for caracter in oracion:
    if caracter.isdigit():
        numeros += 1
    elif caracter == " ":
        espacios += 1
    elif caracter.isalpha():
        letras += 1
        if caracter.lower() in "aeiou":
            vocales += 1
        else:
            consonantes += 1

#resultados
print(f"Números: {numeros}")
print(f"Espacios: {espacios}")
print(f"Vocales: {vocales}")
print(f"Consonantes: {consonantes}")
print(f"Letras totales: {letras}")
print(f"Caracteres totales: {len(oracion) - espacios}")